src/hooks/useStacApi.ts

```diff
@@ -1,38 +1,58 @@
-import { useEffect, useState } from 'react';
+import { useContext, useEffect, useState } from 'react';
 import StacApi, { SearchMode } from '../stac-api';
 import { Link } from '../types/stac';
 import { GenericObject } from '../types';
+import { StacApiContext } from '../context/context';
 
 type StacApiHook = {
-  stacApi?: StacApi
-}
+  stacApi?: StacApi;
+};
+
+function useStacApi(url?: string, options?: GenericObject): StacApiHook {
+  // The context uses this hook to get the StacApi instance.
+  // If a URL is provided, it will create a new StacApi instance.
+  // If no URL is provided, it will use the StacApi instance from the context.
+  // This allows the hook to be used in different contexts, such as in using the
+  // existing instance or to create a new one.
+  const { stacApi: ctxStacApi } = useContext(StacApiContext);
```

Comment:
I don't understand why we need this. `useStacApi` was an undocumented feature, it wasn't meant to be used by implementers. How will this be used?

Reply:
It is basically to access the underlying api instance to make custom requests not covered by the hooks. In this case I'm using it to make the requests needed to update/create collections. Instead of making a call directly with `window.fetch` I'm getting the api instance to make use of its fetch method which gets the url from context and also has nice error handling.

Reply:
The `stacApi` object is exposed through the `StacApiContext`. Would something like this not work?

```js
const { stacApi } = useStacApiContext();
stacApi.fetch("/collections", {
  method: "POST"
  payload: { ... }
  headers: { ... }
})
```

Reply:
Yup, that would work as well. However the `useStacApiContext` is not exported, and having to change something, it made more sense to me that the `useStacApi` hook (which is exported), would return the instance in the context. https://github.com/developmentseed/stac-react/blob/a6194651ec9e7f7b3b6c87193fa292924598df43/src/index.ts#L8-L15

Reply:
Yes ok, in that case I would actually suggest to not export `useStacApi` (which is only exported because it was needed in previous versions and I forgot to remove the export) and to export `useStacApiContext` instead. It makes sense for exactly the developer case you're facing, where you need access to the `stacApi` instance without the need to change `useStacApi` and introducing complexity by making the `url` param optional.